src/node.cpp

```diff
@@ -39,15 +39,57 @@ NewNode::NewNode()
     source = NEW_NODE;
     ovhandle = UNDEF;
     uploadhandle = UNDEF;
-    localnode = NULL;
-    fileattributes = NULL;
+    localnode = nullptr;
+    fileattributes = nullptr;
 }
 
 NewNode::~NewNode()
 {
     delete fileattributes;
 }
 
+NewNode::NewNode(NewNode&& o)
+    : NewNode() // assign defaults
+{
+    swap(std::move(o));
+}
+
+NewNode& NewNode::operator=(NewNode&& o)
+{
+    if (this != &o)
+    {
+        if (attrstring)
+        {
+            delete attrstring;
+            attrstring = nullptr;
+        }
+        localnode = nullptr;
+        if (fileattributes)
+        {
+            delete fileattributes;
+            fileattributes = nullptr;
+        }
+        swap(std::move(o));
+    }
+    return *this;
```

Comment:
if we use unique_ptr we don't need to write this function at all, it's provided automatically by the compiler